packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -243,14 +213,18 @@ export class IndexedDbMutationQueue implements MutationQueue {
     localWriteTime: Timestamp,
     mutations: Mutation[]
   ): PersistencePromise<MutationBatch> {
-    return this.getNextBatchId(transaction).next(batchId => {
+    return this.getMutationQueueMetadata(transaction).next(metadata => {
+      const batchId = metadata.nextBatchId;
+      ++metadata.nextBatchId;
```

Comment:
Hrm...  metadata is per-user and we only populate metadata.nextBatchId with the globally-correct nextBatchId on startup...  after that we just increment it.  So if another tab managed to log in as a different user and create a batch, then we could end up with duplicate batchIds.  I think this is unlikely because auth tries to synchronize auth state across tabs, but I'm not sure we should rely on that.

That said, I don't think we _actually_ need batchIds to be unique across users.  The original intention was to keep them unique so that if the SDK was ever able to send mutations to the server for a non-logged-in user (e.g. the anonymous user, see b/33437802) then we could still send them in the correct order, across users.

So, uh I think we should probably either give up on globally-unique batch Ids or else we need to have a globalNextBatchId stored in IndexedDb somewhere (a new globals store with a single row or something) and use that instead of the per-user nextBatchId.  And in that case we could stop doing the loadNextBatchIdFromDb() thing in start() [and do it once on schema migration instead]...